src/linter.js

```diff
@@ -51,6 +58,9 @@ export default function linter(options, compilation) {
    * @param {string | string[]} files
    */
   function lint(files) {
+    for (const file of asList(files)) {
+      delete crossRunResultStorage[file];
+    }
```

Comment:
Why we need manually delete it? When you remove something from WeakMap often means that you are using it out of place

Reply:
Thats not the WeakMap. The WeakMap keeps a compiler instance to (file=>lintresult map) mapping.

Reply:
If we lint a file and it does not contain errors, we don't want the old result just in case we do not get a result back from the linter.

Reply:
Why we need to keep old result?

Reply:
That's the point of #44 ... we need to keep reporting results until they are resolved. In watch mode if the user edits a file we don't want to discard all the other file results and only show the one file's results.

Reply:
In each run of the compiler, we only get the results of the files we lint... in watch mode, that may only be one file. But this should be cumulative not just the last run.

Reply:
Thanks for explanation